python_modules/dagster/dagster/_core/definitions/data_version.py

```diff
@@ -44,158 +45,177 @@ def foo(x):
             cls,
             code_version=check.str_param(code_version, "code_version"),
-            input_logical_versions=check.mapping_param(
-                input_logical_versions,
-                "input_logical_versions",
+            input_data_versions=check.mapping_param(
+                input_data_versions,
+                "input_data_versions",
                 key_type=AssetKey,
-                value_type=LogicalVersion,
+                value_type=DataVersion,
             ),
         )
 
     @staticmethod
-    def from_tags(tags: Mapping[str, str]) -> Optional[LogicalVersionProvenance]:
+    def from_tags(tags: Mapping[str, str]) -> Optional[DataProvenance]:
         from dagster._core.definitions.events import AssetKey
 
-        code_version = tags.get(CODE_VERSION_TAG_KEY)
+        code_version = tags.get(CODE_VERSION_TAG)
         if code_version is None:
             return None
-        start_index = len(INPUT_LOGICAL_VERSION_TAG_KEY_PREFIX) + 1
-        input_logical_versions = {
-            AssetKey.from_user_string(k[start_index:]): LogicalVersion(tags[k])
+        input_data_versions = {
+            # Everything after the 2nd slash is the asset key
+            AssetKey.from_user_string(k.split("/", maxsplit=2)[-1]): DataVersion(tags[k])
             for k, v in tags.items()
-            if k.startswith(INPUT_LOGICAL_VERSION_TAG_KEY_PREFIX)
+            if k.startswith(INPUT_DATA_VERSION_TAG_PREFIX)
+            or k.startswith(_OLD_INPUT_DATA_VERSION_TAG_PREFIX)
         }
-        return LogicalVersionProvenance(code_version, input_logical_versions)
+        return DataProvenance(code_version, input_data_versions)
+
+    @property
+    @deprecated
+    def input_logical_versions(self) -> Mapping["AssetKey", DataVersion]:
+        return self.input_data_versions
 
 
 # ########################
 # ##### TAG KEYS
 # ########################
 
-LOGICAL_VERSION_TAG_KEY: Final[str] = "dagster/logical_version"
-CODE_VERSION_TAG_KEY: Final[str] = "dagster/code_version"
-INPUT_LOGICAL_VERSION_TAG_KEY_PREFIX: Final[str] = "dagster/input_logical_version"
-INPUT_EVENT_POINTER_TAG_KEY_PREFIX: Final[str] = "dagster/input_event_pointer"
+DATA_VERSION_TAG: Final[str] = "dagster/data_version"
+_OLD_DATA_VERSION_TAG: Final[str] = "dagster/logical_version"
+CODE_VERSION_TAG: Final[str] = "dagster/code_version"
+INPUT_DATA_VERSION_TAG_PREFIX: Final[str] = "dagster/input_data_version"
+_OLD_INPUT_DATA_VERSION_TAG_PREFIX: Final[str] = "dagster/input_logical_version"
+INPUT_EVENT_POINTER_TAG_PREFIX: Final[str] = "dagster/input_event_pointer"
+
+
+def read_input_data_version_from_tags(
+    tags: Mapping[str, str], input_key: "AssetKey"
+) -> Optional[DataVersion]:
+    value = tags.get(
+        get_input_data_version_tag(input_key, prefix=INPUT_DATA_VERSION_TAG_PREFIX)
+    ) or tags.get(get_input_data_version_tag(input_key, prefix=_OLD_INPUT_DATA_VERSION_TAG_PREFIX))
+    return DataVersion(value) if value is not None else None
```

Comment:
Ah I see we are doing that here. I guess I just need to see that this is validated in the test plan.

Reply:
I added a test that accesses legacy tags